### ISSUE TITLE
Metadata retrieval when metadata for one file references additional files (& their metadata)

### ISSUE THREAD

**Issue by JeniT** [w3c/csvw#312]
See #293.

Say you start with `A.csv`. You retrieve its associated metadata at `A.csv-metadata.json` and that file lists two other CSV files, `B.csv` and `C.csv`. There are foreign key references between these resources, so it's necessary to understand the metadata about `B.csv` and `C.csv`. To get the full picture, you need to get hold of `B.csv-metadata.json` and `C.csv-metadata.json` which provide more detail than is in `A.csv-metadata.json`. Each of those can similarly reference other files, `D.csv` and `E.csv` and so on it goes.

A similar scenario plays out if you start with a metadata file and then go to retrieve all the metadata from all the CSV files that are referenced in that metadata file.

There are several issues with this:
- the huge explosion of files that need to be retrieved to process the single CSV file
- the order in which the metadata files from different locations are merged
- the ability of any metadata file to affect the original processing of `A.csv`, which means you really have to get all of them before you can start

I think the only realistic option is to say that when you start from a CSV file, you only get the metadata associated with that CSV file (aside from the user-defined and embedded metadata that is). And that when you start with a metadata file, that's the only metadata you use (it acts as user-defined metadata and you don't then retrieve metadata through the other mechanisms).

**Comment by iherman** [comment]
Oops, @JeniT we acted in parallel and this is a duplicate of #314 ... Is there a way in github to 'merge' issues?

**Comment by JeniT** [comment]
Closing as duplicate.
